### PR TITLE
8286176: Add JNI_VERSION_19 to jni.h and JNI spec

### PR DESCRIPTION
**Pull request by AlanBateman**
JNI is updated in Java 19 so we need to define JNI_VERSION_19 and change GetVersion to return this version.

test/hotspot/jtreg/native_sanity/JniVersion.java is updated to check that JNI_VERSION_19 is returned. The native library in the JMX agent, and several tests, define JNI_OnLoad that return JNI_VERSION_10 as the JNI version needed. These are updated to return JNI_VERSION_19 although these updates aren't strictly needed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8286176](https://bugs.openjdk.org/browse/JDK-8286176): Add JNI_VERSION_19 to jni.h and JNI spec
 * [JDK-8288283](https://bugs.openjdk.org/browse/JDK-8288283): Add JNI_VERSION_19 to jni.h and JNI spec (**CSR**)

### Reviewers
 * [Daniel D. Daugherty](https://openjdk.java.net/census#dcubed) (@dcubed-ojdk - **Reviewer**)
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**)
 * [Mandy Chung](https://openjdk.java.net/census#mchung) (@mlchung - **Reviewer**)
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk19 pull/7/head:pull/7` \
`$ git checkout pull/7`

Update a local copy of the PR: \
`$ git checkout pull/7` \
`$ git pull https://git.openjdk.org/jdk19 pull/7/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7`

View PR using the GUI difftool: \
`$ git pr show -t 7`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk19/pull/7.diff">https://git.openjdk.org/jdk19/pull/7.diff</a>

</details>

### CODE REVIEW

**Review by irisclark — APPROVED** [review]
Corresponding CSR also Reviewed.

**Review by mlchung — APPROVED** [review]

File: test/micro/org/openjdk/bench/java/lang/foreign/libQSortJNI.c [hunk 1]
```diff
@@ -68,7 +68,7 @@ int java_cmp(const void *a, const void *b) {
    int v2 = *((int*)b);
 
    JNIEnv* env;
-   (*VM)->GetEnv(VM, (void**) &env, JNI_VERSION_10);
+   (*VM)->GetEnv(VM, (void**) &env, JNI_VERSION_19);
```
Comment (dholmes-ora) [inline test/micro/org/openjdk/bench/java/lang/foreign/libQSortJNI.c:71]: This seems unnecessary as there is no dependency on version 19 APIs.

**Review by dholmes-ora — APPROVED** [review]
I'm approving because the changes are not incorrect, but a number of them seem unnecessary.

Thanks.

**Comment by dholmes-ora** [comment]
@AlanBateman  sorry I missed your statement "although these updates aren't strictly needed".

**Comment by AlanBateman** [comment]
> @AlanBateman sorry I missed your statement "although these updates aren't strictly needed".

Right, I decided it would be better to just update them rather than leaving them at _10. For the management agent then JNI_OnLoad returning any supported version will do as we build/test is separating to the rest of the JDK. Same thing for the tests. Someone back porting changes might have to adjust of course but the isn't important here.

**Pull request closed without merging**